compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/ReachabilityAnalyzer.java

```diff
@@ -179,6 +182,11 @@ public void visit(BLangLock lockNode) {
     @Override
     public void visit(BLangSimpleVariableDef varDefNode) {
         checkStatementExecutionValidity(varDefNode);
+
+        BLangExpression expr = varDefNode.var.expr;
+        if (expr != null && expr.getKind() == NodeKind.DO_ACTION) {
+            analyzeReachability(expr, env);
+        }
```

Comment:
Alternatively we could implement empty visit methods for all the other kinds of expressions/actions. Not sure if we need it though since reachability applies to pretty much none of them.

Anonymous function expressions seem to be handled separately.

Reply:
We should be able to utilize new visitors for this.